### DISCUSSION TITLE
Randomly combine (transparent) png

### DISCUSSION THREAD

**Discussion by PaoloPgn** [ImageMagick/ImageMagick#4989] [Digital Image Processing]
Hello,
I got 3 folders (01, 02, 03) with backgrounds (1) heads (2) body (3). I need to **randomly combine** this 3 elements but, _backgrounds_ have to be _the first one_, _heads and body_ (transparent png) have to be over the background **leaving the transparency** of the other 2 images (head and body) **intact**.
And I need to save the result (as png) in the folder (4) of the 3 elements randomly combined together.
Is that possible?
Thank you

**Comment by fmw42** [comment]
Yes with some scripting to access the images from the various folders.

`convert background body head -background transparent -flatten result`

should do the image combining.

What is your platform/OS?  Do you have ImageMagick installed? If so, what version?

**Reply by PaoloPgn** [reply]
Thank you a lot for your answer. I'm on a Mac with Homebrew installation of the last build. I'll try this. (sorry I'm at the very beginning so I think I have to made some test before find what I'm looking for)

**Comment by fmw42** [comment]
For new users, see

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli
https://legacy.imagemagick.org/discourse-server/index.php


For flatten, see

https://imagemagick.org/script/command-line-options.php#flatten
https://legacy.imagemagick.org/Usage/layers/#flatten
